### PR TITLE
Input Sharding and Precision Debugging 

### PR DESCRIPTION
**Pull request by chelseajohn**
This PR fixes the following:
- Make single eval run all mini-app configs
- For the Vandermonde matrix formation, when doing TP>1 the matrix was normalized using local min/max positions (meaning current slice of particles min and max, which is not always same as min and max of full particles) this was creating deviations so changed it to global min and max normalization.
- Implemented possibility of specifying Float64 and Float32 precision for P & Q layers and FNO_DSE layer (including VandermondeTransformMatrixFree)

### CODE REVIEW

File: operator_learning/layers/dse.py [hunk 1]
```diff
@@ -1,17 +1,28 @@
...
+    ContiguousGrad
+)
 
 class SpectralConv_dse(nn.Module):
     def __init__(self, dv, kX, kY=None, kZ=None, dataClass='pic', bias=False, 
-                 dim=1, use_complex_amp=False, tp_mesh=None):
+                 dim=1, use_complex_amp=False, tp_mesh=None, dtype=torch.complex64):
+        
+        """
+        Spectral weights R and Bias must be kept in Float64 when particle
```
Comment (srikrrish) [inline operator_learning/layers/dse.py]: Based on your recent results it is not a must right?

File: operator_learning/layers/mlp.py [hunk 2]
```diff
@@ -61,30 +64,37 @@ def __init__(self,
 
             if mode == 'conv':
                 Conv = nn.Conv2d if n_dims == 2 else nn.Conv3d
-                layer = Conv(in_ch, out_ch, kernel_size=1)
+                layer = Conv(in_ch, out_ch, kernel_size=1, bias=True)
```
Comment (srikrrish) [inline operator_learning/layers/mlp.py:67]: Why do we now specify `bias=True` was it making any difference?

File: operator_learning/layers/dse.py (hunk 1 above)
Comment (chelseajohn) [reply operator_learning/layers/dse.py]: yes, forgot to remove it. I will commit the change

File: operator_learning/layers/mlp.py (hunk 2 above)
Comment (chelseajohn) [reply operator_learning/layers/mlp.py:67]: The default is True (https://docs.pytorch.org/docs/2.13/generated/torch.nn.Conv1d.html), I just explicitly added it during debugging.

**Review by srikrrish — APPROVED** [review]
